xwiki-platform-core/xwiki-platform-skin/xwiki-platform-skin-api/src/main/java/org/xwiki/skin/Resource.java

```diff
@@ -65,4 +68,18 @@
      * @throws Exception when failing to create a URL for the resource
      */
     String getURL(boolean forceSkinAction) throws Exception;
+
+    /**
+     * Create a URL for the resource.
+     *
+     * @param forceSkinAction true if a dynamic skin URL should be forced
+     * @param queryParameters add the following parameters to the URL
+     * @return a URL to the resource
+     * @throws Exception when failing to create a URL for the resource
+     * @since 11.0RC1
+     */
+    @Unstable
+    default String getURL(boolean forceSkinAction, Map<String, String> queryParameters) throws Exception {
```

Comment:
It's not very clear to me why this `queryParameters` is needed. You are not really supposed to know which parameter to add when you call getURL. The whole idea is that only the Resource implement knows what makes this specific resource unique (document version for an object, attachment version for an attachment, date or whatever for a file on disk, etc.) and will automatically add it to the generated URL.